### PR TITLE
Issue #11214: Specified violation message in EmptyBlockCheck

### LINKED ISSUE #11214: Specify violation messages in input files
Single Check might have multiple violations messages, we need to see exact violation message if Check has multiple messages, to be sure that violation of correct type is placed at right line.

Violation messages have not been specified in certain input files, a list of checks in which violation message is not specified in the input file and have more than one violation message key has been specified here-
https://github.com/checkstyle/checkstyle/blob/72f182241a3b8a6d905328dfcfc9739dfffac9bd/src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java#L99

To Do-

- Pick a check (single Check Inputs files update in single Pull Request) To allocate Check update for you please make comment in issue "I am on XxxxxCheck"
- Find all input files that use that check
- Add violation message at the end of the `// violation` comment
- Remove the check from the suppression list

Same has been demonstrated at 461cd048bb059ea6948945104682fb99d4e155dc
If the violation message contains special regex characters, escape that character.
Ignore input files using more than a single check.

Ignore lines with multiple violations, eg-
```
// 2 violations
// 2 violations above
// 4 violations below
```
and any other violation message of this nature.

List of Checks: 

- [x]   RegexpCheck
- [x]   EmptyForInitializerPadCheck
- [ ]   JavadocStyleCheck
- [x]   SummaryJavadocCheck

Old tracker (some might be outdated):

- [x] RightCurlyCheck
- [x] EqualsAvoidNullCheck
- [x] MethodParamPadCheck
- [x] DescendantTokenCheck
- [x] RequireThisCheck
- [x] EqualsHashCodeCheck
- [x] JavaNCSSCheck
- [x] ReturnCountCheck
- [x] AnnotationUseStyleCheck
- [x] EmptyLineSeparatorCheck
- [x] OrderedPropertiesCheck
- [x] AbstractJavadocCheck
- [x] DeclarationOrderCheck
- [x] MissingOverrideCheck
- [x] AbstractClassNameCheck
- [x] MethodCountCheck
- [x] PackageDeclarationCheck
- [x] JavadocTypeCheck
- [x] LeftCurlyCheck
- [ ] RegexpCheck
- [x] WhitespaceAfterCheck
- [x] MultipleVariableDeclarationsCheck
- [x] EmptyForInitializerPadCheck
- [x] UnnecessaryParenthesesCheck
- [x] JavadocMethodCheck
- [x] ModifierOrderCheck
- [x] MissingDeprecatedCheck
- [x] RegexpOnFilenameCheck
- [x] RedundantImportCheck
- [x] EmptyBlockCheck
- [x] UniquePropertiesCheck
- [x] WhitespaceAroundCheck
- [x] OperatorWrapCheck
- [x] GenericWhitespaceCheck
- [ ] EmptyForIteratorPadCheck
- [ ] JavadocStyleCheck
- [x] JavadocPackageCheck
- [x] FallThroughCheck
- [x] WriteTagCheck
- [x] TranslationCheck
- [x] FileTabCharacterCheck
- [x] ImportOrderCheck
- [x] JavadocContentLocationCheck
- [x] JavadocParagraphCheck
- [ ] SummaryJavadocCheck
- [x] DefaultComesLastCheck
- [x] AnnotationLocationCheck
- [ ] CustomImportOrderCheck
- [x] NewlineAtEndOfFileCheck
- [x] VariableDeclarationUsageDistanceCheck
- [x] ImportControlCheck

format it like:
https://github.com/checkstyle/checkstyle/blob/5b2a0efc8c577d6382effe818b0d94d489397352/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/declarationorder/InputDeclarationOrderOnlyModifiers.java#L16

if message is long, we can shorten it by regexp usage (but do not over use regexp, message should be readable):
https://github.com/checkstyle/checkstyle/blob/4bd9856de41ae74c903bf8468b3355ded0b73a58/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/equalsavoidnull/InputEqualsAvoidNullEnhancedInstanceof.java#L15

There are a lot of examples in code just search for `// violation` in Input files for more examples.

### PR DESCRIPTION
**Pull request by SarveshLimaye**
Issue #11214 
Specified violation message in EmptyBlockCheck.

### CODE REVIEW

**Comment by Vyom-Yadav** [comment]
Follow proper commit message, this is not a `minor` PR, specify issue in commit message as well as PR desc.
```
Proper commit message should adhere to the following rules:
    1) Must match one of the following patterns:
        ^Issue #\d+: .*$
        ^Pull #\d+: .*$
        ^(minor|config|infra|doc|spelling|dependency|supplemental): .*$
    2) It contains only one line of text
    3) Must not end with a period, space, or tab
    4) Commit message should be less than or equal to 200 characters
```
Make sure `mvn clean verify` passes successfully on local.

**Review by rnveach — CHANGES_REQUESTED** [review]
@SarveshLimaye This is not a minor commit, and this does not fix the issue as the wording in the first post says as there are multiple checks involved in the issue.

Commit message must match the issue paradigm.

All this needs to be fixed and CI needs to pass.

**Comment by romani** [comment]
@SarveshLimaye , review CI problems and resolve them.
before push make sure that 'mvn verify'  are passing to you local.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockCase.java [hunk 1]
```diff
@@ -13,13 +13,13 @@
     void method1(int a) {
         switch (a) {}
         switch (a) {case 1: ; } // ok
-        switch (a) {case 1:{}}  // violation
+        switch (a) {case 1:{}}  // violation ,Empty block
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockCase.java:16]: remove ','
format it like:
https://github.com/checkstyle/checkstyle/blob/5b2a0efc8c577d6382effe818b0d94d489397352/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/declarationorder/InputDeclarationOrderOnlyModifiers.java#L16

**Review by romani — CHANGES_REQUESTED** [review]
item:

**Comment by SarveshLimaye** [comment]
@romani Thank you so much for the review . I made all the suggested changes but I am still getting some errors.

image

**Comment by romani** [comment]
@SarveshLimaye , do not hesitate to debug a problem in your IDE to see what test is expecting

**Comment by SarveshLimaye** [comment]
Hey @romani , I have solved all the previous errors and Emptyblock check test is passing but I am getting this errors after running mvn clean verify .

image

image

image

Could you pls guide me further.

**Comment by romani** [comment]
It might be problem with line ending of your sources.
You might be on windows, so you need to do
https://github.com/checkstyle/checkstyle/blob/890e1f4ab44a2950b58af3a9fb976a8c21e6eff1/appveyor.yml#L23
Before cloning of checkstyle repository

**Comment by SarveshLimaye** [comment]
@romani Yes I am windows . I have already cloned the repo  and raised pr . Should I run this command now??

**Comment by romani** [comment]
Read https://stackoverflow.com/a/12254105

Make sure you have right config to clone. It might be set globally, so you need to change it before checkstyle clone.

**Comment by SarveshLimaye** [comment]
@romani 
image

Still I am getting same error

**Comment by SarveshLimaye** [comment]
image

@romani  core.autocrlf is true for local repo still I am facing same error.

**Comment by romani** [comment]
@rnveach , can you help ? I do not have windows.

**Comment by rnveach** [comment]
@romani We removed the core.autocrlf instructions for windows users because we said its not needed anymore. If we find this isn't working, then we need to revert our changes associated to removing it.

**Comment by rnveach** [comment]
@SarveshLimaye I don't work with the command line very much, as I use git through Eclipse.

If you enabled `core.autocrlf` after you cloned checkstyle, you will need to delete the files cloned (everything except `.git` and any metadata folders from your IDE), and then you will need to bring them back by either reverting them or doing a hard reset.

Once the files are back, this will not fix commits you have already created for this PR or any other. Some IDEs have commands to do this conversion for you. You can start there on all the files you modified and ammending the commits for the updates it makes.

So you are aware `core.autocrlf` can be set for just Checkstyle's project and not globally. I don't remember the command line to set it for just for 1 project, but it can be done by modifying the `.git\config` file so it has the following:
````
[core]
	autocrlf = true
````

**Comment by romani** [comment]
@strkkk or @pbludov , do you have access to windows os to recheck how it works?

I remember we got extra CI in windows, it is not only appveyor.

**Comment by strkkk** [comment]
I have `core.autocrlf=true` on windows machine and have no issues.

There is also useful checkbox in IDEA settings
изображение

**Comment by rnveach** [comment]
I haven't really seen this message before but from I am seeing it sounds ok as just a warning.

Basically `autocrlf` turns files on your local to CRLF, but when it comes time to committing the files to git, it swaps them back to LF and stores them that way in the repository. Really the only way you can confirm how line endings are stored then is by examining the repository with autocrlf off.

**Comment by rnveach** [comment]
@romani It seems we need to restore documentation on autocrlf. It is also strange appveyor does not fail as I believe we removed it there too.
We will have to restore it.

**Comment by romani** [comment]
I am ok to restore. Usage on windows should be easy and well documented

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockSemanticText.java [hunk 2]
```diff
@@ -84,7 +84,7 @@ synchronized void foo() {
     static {
 
         int a = 0;}
-    // violation below, Empty block
+    // violation below
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockSemanticText.java:87]: Do you see a reason why messages are supported for violation above/below ?

**Review by romani — CHANGES_REQUESTED** [review]
Items:

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockSemanticText.java (hunk 2 above)
Comment (romani) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockSemanticText.java:87]: Example https://github.com/checkstyle/checkstyle/pull/11051/files#diff-52fde5cee31a03e135e9c918bdde51ab828aa8d2a4fd8cc9a74f9f651985961eR42

Comment (SarveshLimaye) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockSemanticText.java:87]: Yeah I will specify the violation msg for the above/below

**Comment by SarveshLimaye** [comment]
@romani  even though my commit msg matches the paradigm I am receiving this error after running mvn clean verify

image

**Comment by rnveach** [comment]
@SarveshLimaye Commit message doesn't match. There must be a space after the `:`. Look closely at the regexp for the rule broken, number 1.

**Comment by SarveshLimaye** [comment]
@rnveach Thank you so much it solved the failure . Now I am getting these 2 errors.  Could you pls help me
image

**Comment by romani** [comment]
update inputs to be less than 100 symbols in size

**Comment by SarveshLimaye** [comment]
@romani  How do I shorten it below 100. Could you pls guide me

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockDefault2.java [hunk 3]
```diff
@@ -43,21 +43,21 @@ void method2(int a) {
...
+            default: {} method2(a);     // violation 'Must have at least one statement'
         }
 
         switch (b) {
             case 2: break;
             default: method2(b); {}     // ok
         }
 
-        switch (a+b) {case 1: break; default: {} ; }    // violation
+        switch (a+b) {case 1: break; default: {} ; }    // violation 'Must have at least one statement'
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockDefault2.java:54]: remove extra spaces before `// violation`

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockDefault2.java (hunk 3 above)
Comment (SarveshLimaye) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockDefault2.java:54]: @romani  Done . Similary can you tell me how to shorten it in InputEmptyBlockSemantic2Statement file.

**Comment by SarveshLimaye** [comment]
image
 
Getting a weird build error now. Could you pls help @rnveach @romani

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockDefault2.java (hunk 3 above)
Comment (romani) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockDefault2.java:54]: `System.identityHashCode("a")` replace with any other statement like `a++;`

**Comment by romani** [comment]
>Getting a weird build error now

Not a weird, it just tell you that checkstyle execution on its own code is failing. Scroll up to see actual errors.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockDefault2.java (hunk 3 above)
Comment (romani) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockDefault2.java:54]: Anything that is not about blocks can be changed.

**Comment by SarveshLimaye** [comment]
image

Dont know why ci is failing here mvn clean verify was passing successfully

**Comment by romani** [comment]
Please do effort to read all CI failures.
https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=7901&view=results

Exact reason for one of problem:
https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=7901&view=logs&j=c902ebb4-c9f8-5f09-4e17-ff78fbbc842e&t=9ca98c81-ff64-58f0-9d03-a23ac1c4a111&l=776

**Comment by rnveach** [comment]
@SarveshLimaye Please avoid creating any merge commits.
https://github.com/checkstyle/checkstyle/wiki/PR-rules , see item 2

**Comment by romani** [comment]
@SarveshLimaye, ping, please finish PR

**Comment by rnveach** [comment]
I updated the PR to finish this.

**Review by romani — APPROVED** [review]

**Comment by Shaman20** [comment]
> image

@SarveshLimaye Hey I am facing the same kind of error in my check #11645. Could you please tell me how you resolved this error?

**Pull request merged**